client/core/core.go

```diff
@@ -1629,6 +1629,80 @@ func (c *Core) WalletSettings(assetID uint32) (map[string]string, error) {
 	return dbWallet.Settings, nil
 }
 
+func (c *Core) setAppPass(pw []byte) (encrypt.Crypter, error) {
+	if len(pw) == 0 {
+		return nil, fmt.Errorf("empty password not allowed")
+	}
+
+	crypter := c.newCrypter(pw)
+	err := c.db.Store(keyParamsKey, crypter.Serialize())
+	if err != nil {
+		return nil, fmt.Errorf("error storing key parameters: %w", err)
+	}
+
+	return crypter, nil
+}
+
+func (c *Core) updateWalletsEncPW(oldCrypter, newCrypter encrypt.Crypter) error {
+	c.walletMtx.Lock()
+	defer c.walletMtx.Unlock()
+	for _, w := range c.wallets {
+		// If encrypted password set, decrypt using old app pw then encrypt again
+		// using new app pw.
+		if len(w.encPW) != 0 {
+			// Decrypt with old app pw.
+			pwB, err := oldCrypter.Decrypt(w.encPW)
+			if err != nil {
+				return err
+			}
+			// Re-encrypt with new app pw.
+			encPW, err := newCrypter.Encrypt(pwB)
+			if err != nil {
+				return err
+			}
+			// Store encrypted pw.
+			w.encPW = encPW
+			err = c.db.SetWalletPassword(w.dbID, w.encPW)
+			if err != nil {
+				return codedError(dbErr, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// ChangeAppPass updates the application password to the provided new password
+// after validating the current password.
+func (c *Core) ChangeAppPass(appPW, newAppPW []byte) error {
+	// Validate current password.
+	oldCrypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return newError(authErr, "ChangePassApp password error: %v", err)
+	}
+	// Set new password.
+	newCrypter, err := c.setAppPass(newAppPW)
+	if err != nil {
+		return err
+	}
+	// Update xcWallets' encrypted passwords.
+	c.updateWalletsEncPW(oldCrypter, newCrypter)
+	// Update dexConnections' dexAccount private & encrypted key.
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	for _, dc := range c.conns {
+		// Generate new dexAccount private key and encrypt it.
+		dc.acct.generatePrivateAndEncKeys(newCrypter)
```

Comment:
We don't want to generate new private keys, because the existing public-private key pair is the account identity;instead we need to re-encrypt the privatekey.

Reply:
Sweet, will do.
Thanks @chappjc appreciate your insights and input! 🥂

Reply:
@chappjc Updated and ready for a serious review.